File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        count1 = 0
        count2 = 0
        res = ""
        temp = 0
        for i,c in enumerate(s):
            if c == "(":
                count1 += 1
            elif c == ")":
                count2 += 1
            if count1 == count2:
                res += s[temp+1:i]
                temp = i + 1
        return res
        

File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        int count1 = 0;
        int count2 = 0;
        int temp = 0;
        String res = "";
        for(int i=0;i<s.length();i++){
            char str1 = s.charAt(i);
            if(str1 == '('){
                count1 = count1 + 1;
            }
            else if (str1 == ')')
            {
                count2 = count2 + 1;
            }
            
            if(count1 == count2){
                res = res + s.substring(temp + 1, i);
                temp = i + 1;
            }
            
        }
        return res;
        
    }
}